listenbrainz/db/external_service_oauth.py

```diff
@@ -6,7 +6,7 @@
 
 
 def save_token(user_id: int, service: ExternalServiceType, access_token: str, refresh_token: str,
-               token_expires_ts: int, record_listens: bool, scopes: List[str]):
+               token_expires_ts: int, record_listens: bool, scopes: List[str], external_user_id: str = None):
```

Comment:
Passing-by review :)
the type hint is wrong, it should be an option since the default argument is `None`
```suggestion
               token_expires_ts: int, record_listens: bool, scopes: List[str], external_user_id: Optional[str] = None):
```
The type is not in the scope, so you have to update imports at the top as well:
`from typing import List, Optional, Union`